confgenerator/resourcedetector/gce_detector.go

```diff
@@ -0,0 +1,125 @@
+package resourcedetector
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Keep synced with the Prometheus repo
+const (
```

Comment:
I would suggest keeping this detector generic and leave all the translation to the prometheus style labels to the `prometheus` receiver. This detector is useful for other components too. One such example is: [b/219991181](http://b/219991181) | P3 | Allow LogName to be used in processors